src/plugins/data_explorer/public/components/sidebar/index.tsx

```diff
@@ -24,23 +24,29 @@ export const Sidebar: FC = ({ children }) => {
 
   const {
     services: {
-      data: { indexPatterns, dataSources, ui },
+      data: {
+        indexPatterns,
+        dataSources,
+        ui,
+        query: { queryString },
+      },
       notifications: { toasts },
       application,
     },
   } = useOpenSearchDashboards<DataExplorerServices>();
 
   useEffect(() => {
-    const subscriptions = ui.Settings.getEnabledQueryEnhancementsUpdated$().subscribe(
-      (enabledQueryEnhancements) => {
+    const subscriptions = queryString
```

Comment:
i think we can clean more of this file and all we really need it to do is check the ui settings if enhancements enabled. if enhancements enabled don't show the other selector